enhancements/kube-apiserver/enforce-labels.md

```diff
@@ -0,0 +1,145 @@
+
+#### Safeguaring control-plane components
+
+Gatekeeper uses a single admission webhook to for constraints enforcement. We assume the same solution will be used for the mutation feature. The control-plane resources can be excluded from gatekeeper interference by specifying an appropriate namespace selector on the webhook.
+Gatekeeper would be installed/configured by an operator, which would adjust the webhook namespace selector (based on some GatekeeperInstallConfig CRD). 
+
+The webhook namespaceselector allows to filter the included namespaces based on the namespace labels (details available: [here](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/) [here](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/)) .
+
+Also the default gatekeeper webhook configuration is using "failurePolicy​: ​Ignore", we assume this would be extended to the mutation feature (more on this: [here](https://github.com/open-policy-agent/gatekeeper#admission-webhook-fail-open-status)). Any resources which would slip into the system unmodified would later be corrected using the additional relabeling feature (implemented in the second phase).
+
+
+#### Performance
+
+Multi-pod deployments of Gatekeeper is already implemented. This will allow to run multiple pods handling the mutating webhook to allow for better performance, better latency and resilience.
+The relabeling feature would preferably be implemented to run on multiple pods as well (daemonset?). The relabeling feature will most probably be implemented as a controller watching for policy changes, and at system startup to fix any drift created during gatekeeper downtime. Note that the design of the relabeling feature has not started yet, so no details are available as of now.
+
+
+#### Risk assesment
+
+There are multiple risks involved with delivering the feature using gatekeeper:
+
+- ETA: the time frame for the first part of the feature (mutating webhook) is quite tight, even though there are currently 3 people (different projects) commited to work on this feature
+- We have no maintainers on the project, which could be a bottleneck with getting the feature code in, or getting some required features in.
+- The requirements and design of the second part of the feature (drift correction) is nowhere near ready. There are still open questions on how do resolve issues like constraint conflicts or convergence.
+- The upstream code might not be of the required quality by the required time.
+
+#### API
+
+The detailed API is still under discussion by the Gatekeeper community. A design proposal accepted in a general form by the gatekeeper community can be found [here](https://docs.google.com/document/d/1MdchNFz9guycX__QMGxpJviPaT_MZs8iXaAFqCvoXYQ/edit?ts=5f73fb77#).
+Once finalized, there should be little risk of the API changing much after the alpha version.
```

Comment:
I read this in a different way in the referenced doc:

> This custom parser would not be compatible with a Rego-based mutation implementation. This puts it in direct conflict with the De-Risking the Future section of the Mutation Dynamics doc. We are accounting for this by staying in alpha until we get a clear signal that we have taken a good approach.

It means that this is nothing the project commits to *before* moving out of alpha.

Reply:
The project will be kept as dev preview until the API is out of alpha. Tech preview/GA will only be considered when API is beta.
Added a Note here, also updated the graduation criteria